nixos/modules/services/system/hot-resize.nix

```diff
@@ -0,0 +1,84 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.hotResize;
+in
+{
+  options.services.hotResize = {
+    enable = lib.mkEnableOption "Enable hot resize service for filesystems";
+
+    device = lib.mkOption {
+      type = lib.types.str;
+      example = "/dev/sda1";
+      description = "Device to resize";
+    };
+
+    fsType = lib.mkOption {
+      type = lib.types.enum [
+        "ext4"
+        "xfs"
+        "btrfs"
+      ];
+      default = "ext4";
+      description = "Filesystem type (supported: ext4, xfs, btrfs)";
+    };
+
+    mountPoint = lib.mkOption {
+      type = lib.types.str;
+      example = "/";
+      description = "Mount point of the filesystem to resize";
+    };
+  };
```

Comment:
Seems odd that we can only resize a single device at a time. This should probably be a `list of (submodule)`.